plat/kvm/x86/lxboot.c

```diff
@@ -110,14 +114,20 @@ lxboot_init_mem(struct ukplat_bootinfo *bi, struct lxboot_params *bp)
 		if (end <= start)
 			continue;
 
-		mrd.pbase = start;
-		mrd.vbase = start; /* 1:1 mapping */
+		mrd.pbase = PAGE_ALIGN_DOWN(start);
+		mrd.vbase = mrd.pbase; /* 1:1 mapping */
+		mrd.pg_off = start - mrd.pbase;
 		mrd.len = end - start;
+		mrd.pg_count = PAGE_COUNT(mrd.pg_off + mrd.len);
 
 		if (entry->type == LXBOOT_E820_TYPE_RAM) {
 			mrd.type = UKPLAT_MEMRT_FREE;
 			mrd.flags = UKPLAT_MEMRF_READ | UKPLAT_MEMRF_WRITE;
 
+			/* Free memory regions have
+			 * mrd.len == mrd.pg_count * PAGE_SIZE
+			 */
+			mrd.len = PAGE_ALIGN_UP(mrd.len);
```

Comment:
> Length in bytes of the resource inside this region

Oops should have been ` + mrd.pg_off`

Reply:
Ah, wait, nevermind, I remembered. Ok I actually extend free memory regions to always be multiple of PAGE_SIZE. This is to extend to page aligned margins both start/end of free mrd's. Some memory maps give only page-sized free memory regions and the next memory region starts at an in-page offset. This means we may lose some bytes of free memory that may have otherwise been used by allocators that don't require page granularity like bbuddy, e.g. TLSF.